Binary files simple-mtpfs-0.2/.DS_Store and simple-mtpfs-0.2-new/.DS_Store differ
diff -uNr simple-mtpfs-0.2/src/simple-mtpfs-mtp-device.cpp simple-mtpfs-0.2-new/src/simple-mtpfs-mtp-device.cpp
--- simple-mtpfs-0.2/src/simple-mtpfs-mtp-device.cpp	2017-11-07 14:52:36.612104759 +0800
+++ simple-mtpfs-0.2-new/src/simple-mtpfs-mtp-device.cpp	2017-11-07 14:51:46.522529282 +0800
@@ -150,7 +150,7 @@
     // Try to reset USB device, so we don't wait until LIBMTP times out.
     // We do this every time we are about to mount a device, but better
     // connect on first try, than wait for 60s timeout.
-    smtpfs_reset_device(raw_device);
+    //smtpfs_reset_device(raw_device);
 #endif // HAVE_LIBUSB1
 
     // Do not output LIBMTP debug stuff
@@ -196,7 +196,7 @@
     // Try to reset USB device, so we don't wait until LIBMTP times out.
     // We do this every time we are about to mount a device, but better
     // connect on first try, than wait for 60s timeout.
-    smtpfs_reset_device(raw_device);
+    //smtpfs_reset_device(raw_device);
 
     bool rval = connect(raw_device);
 
@@ -245,6 +245,7 @@
     if (LIBMTP_Get_Storage(m_device, LIBMTP_STORAGE_SORTBY_NOTSORTED) < 0) {
         std::cerr << "Could not retrieve device storage.\n";
         std::cerr << "For android phones make sure the screen is unlocked.\n";
+        system("echo For android phones make sure the screen is unlocked >> /tmp/debugmtpdev");
         logerr("Could not retrieve device storage. Exiting.\n");
         LIBMTP_Dump_Errorstack(m_device);
         LIBMTP_Clear_Errorstack(m_device);
